jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFComponentProcessor.java

```diff
@@ -173,6 +176,69 @@ private void processXmlManagedBeans(final DeploymentUnit deploymentUnit, final S
         }
     }
 
+    /**
+     * WFLY-6617 workaround
+     * According to  JSF 2.2 spec, it should be possible to inject beans using @EJB annotation into
+     * PhaseListeners.
+     */
+    private void processPhaseListeners(final DeploymentUnit deploymentUnit, final Set<String> managedBeanClasses) {
```

Comment:
This mostly looks correct, although I don't really like the parser it is probably the best approach.

For most other components we already have parsers so this is not an issue, but it looks like the JSF impl does the parsing itself so that is not an option here.